### PR TITLE
Restore original man page URLs

### PR DESCRIPTION
**Pull request by jmarshall**
Unversioned tool URLs on <http://www.htslib.org> like _…/doc/samtools.html_ have been stable URLs since June 2015 (03194ffa049d) and file format URLs like _…/doc/sam.html_ even longer, since October 2014 (6c9b83b98b25).

These URLs have appeared in various external links over the years, and these links no longer work since the 1.10 release and web site update. For example, see [this pyfaidx bgzip link](https://pypi.org/project/pyfaidx/#support-for-compressed-fasta) (also in [their README](https://github.com/mdshw5/pyfaidx#support-for-compressed-fasta) and various links from forums like biostars such as <https://www.biostars.org/p/330555/> (just search the forum for e.g. `samtools.html` to see lots of such links).

This PR restores these original URLs so that bookmarks and external links continue to work.

### CODE REVIEW

**Comment by daviesrob** [comment]
It looks like this does a bit more than restore the old URLs...

I've been looking into [jekyll-redirect-from](https://github.com/jekyll/jekyll-redirect-from) as a way of fixing this problem.  See for example https://daviesrob.github.io/www.htslib.org/doc/samtools.html There are a few caveats to that method, though:

* It's not a proper http redirect, so it only works in web browsers.

* It doesn't work for the bcftools page as that one doesn't have Jekyll front matter.  That could easily be fixed by manually adding the necessary redirect pages, though.

* I had to remove `url` from the site settings in `_config.yml` to make it work on my Github pages site.  I'm not sure what that will do to the `www.htslib.org` copy.

**Comment by jkbonfield** [comment]
Bah, I forgot to check whether anyone was deep linking into our site when I made that change.  Sorry.

I did do it for a reason though, even if ill advised.  The man page script ought to include the man section as we may at some point have documentation on functions, programs, formats, etc with potential name clashes.  It was wrong in the first place to omit this part of the filename.

I'd be in favour of a redirect approach.

**Comment by jmarshall** [comment]
These issues are not new, and they were considered when the website was set up in 2014/15. The thinking then was:

* URLs lke _…/doc/samtools.html_ are clearer for non-programmers than _…/doc/samtools.1.html_ — is that for samtools v1.0? The clearer simpler URLs were chosen even though they preclude the unlikely scenario of the maintainers writing new man pages with the same name as an existing htslib/samtools/bcftools one in a different section.

* At the time, versioned man pages were at URLs like _…/doc/samtools-1.2.html_ which was further reason for not having spurious “.1”s in the URLs. Now that the samtools man page has been split up, to be sure there are advantages to _…/doc/1.10/samtools*.html_ rather than _…/doc/samtools-*-1.10.html_ though note that the old versioned URLs are currently 404 too (but I'm less concerned about them).

* At such point as further function, format, etc documentation is added to the website, this will most likely be done via Doxygen (or similar) from the function etc documentation in the header files. Doxygen expects to be in control of its output directory and splatters lots of files into it. So the plan for API documentation was to have it in a _…/doc/api/*_ subdirectory — so there is no risk of colliding with existing _…/doc/*_ URLs.

In short, clearer simpler _…/doc/samtools.html_ URLs were actively chosen. A redirect approach would expose the crappier URLs instead. (Recall also that the different _.1_, _.3_ etc man page filename extensions are mostly a historical accident. The real way the section is encoded into the man page's filepath is by the subdirectory it's in: _man/man1/*_ etc.)

Actually I'm having trouble thinking of a samtools scenario in which a name clash between sections wouldn't be best resolved by giving the later man page a better more descriptive name…

**Comment by daviesrob** [comment]
The old links are now back, and I've added redirects to fix the links to the historic man pages.  The redirects mostly work, apart from they annoyingly go back to the non-TLS location (possibly something to bring up with the jekyll-redirect-from people).

I've also taken the liberty of fixing a few other issues, like the missing `samtools coverage` page and a few typos and bits of broken rendering.  I've also added a few updates that appeared since the last release, but are sill relevant to it (like the `--threads` option documentation).  I have not added anything to do with new functionality that appeared after release 1.10, or fixes for problems in the source man pages (like the unexpected boldness at the start of the [`tview` synopsis](https://www.htslib.org/doc/samtools-tview.html#SYNOPSIS).)
